### PR TITLE
[CH-318] Introduce post process for native config to inject some spark config to it 

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?
 Introduce post process for native config to inject some spark config to it

### CODE REVIEW

**Comment by taiyang-li** [comment]
Maybe clickhouse spill settings in https://github.com/Kyligence/ClickHouse/issues/318 can be injected in similar ways.

**Comment by taiyang-li** [comment]
PS, 已有实现
```
基本处理逻辑：
1. 将spark hive相关的config同步到map中
2. 将spark s3相关的config同步到map中
3. 将spark memory相关的config同步到map中 
4. 将spark timezone相关的config同步到map中
5. 将spark backend相关的config同步到map中 
*6*: 将spark中相关的config
```

**Comment by taiyang-li** [comment]
Unrelated error in velox uts  cc @rui-mo 
image

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 1]
```diff
@@ -294,6 +294,10 @@ object GlutenConfig {
   val S3_USE_INSTANCE_CREDENTIALS = "fs.s3a.use.instance.credentials"
   val SPARK_S3_USE_INSTANCE_CREDENTIALS: String = HADOOP_PREFIX + S3_USE_INSTANCE_CREDENTIALS
 
+  /// Spark executor configs
+  val EXECUTOR_PREFIX = "spark.executor"
+  val EXECUTOR_CORES = EXECUTOR_PREFIX + ".cores"
+
```
Comment (zzcclp) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:300]: remove first ?

**Comment by zzcclp** [comment]
please update the pr title and desc.

**Review by jinchengchenghh — APPROVED** [review]

**Comment by taiyang-li** [comment]
> please update the pr title and desc.

done

File: gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala [hunk 2]
```diff
@@ -23,6 +23,9 @@ import org.apache.spark.sql.catalyst.plans.physical.Partitioning
 import org.apache.spark.sql.connector.read.InputPartition
 import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, PartitionDirectory}
 import org.apache.spark.sql.types.StructField
+import org.apache.spark.SparkConf
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/backendsapi/TransformerApi.scala:26]: remove ? it's unused import

File: gluten-data/src/main/scala/io/glutenproject/backendsapi/glutendata/GlutenTransformerApi.scala [hunk 3]
```diff
@@ -77,4 +77,9 @@ abstract class GlutenTransformerApi extends TransformerApi with Logging {
                            selectedPartitions: Array[PartitionDirectory]): Seq[InputPartition] = {
     InputPartitionsUtil.genInputPartitionSeq(relation, selectedPartitions)
   }
+
+  def postProcessNativeConfig(nativeConfMap: util.Map[String, String],
```
Comment (zzcclp) [inline gluten-data/src/main/scala/io/glutenproject/backendsapi/glutendata/GlutenTransformerApi.scala:81]: there is missing `import java.util`

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by rui-mo — APPROVED** [review]

**Pull request merged**